models/ssh_key.go

```diff
@@ -543,16 +545,38 @@ func RewriteAllPublicKeys() error {
 		_, err = f.WriteString((bean.(*PublicKey)).AuthorizedString())
 		return err
 	})
-	f.Close()
+
 	if err != nil {
+		f.Close()
```

Comment:
you can still defer `f.Close()`, it will just not do anything if it's already closed :)